client/lib/promote-post/index.ts

```diff
@@ -206,6 +212,32 @@ export const requestDSP = async < T >(
 	}
 };
 
+const handleDSPError = async < T >(
+	error: any,
```

Comment:
Yes! I'll ad a proper type. Thanks!

Reply:
Ah yes, I remember, I added it then reverted because of this:
<img width="813" alt="image" src="https://github.com/Automattic/wp-calypso/assets/43957544/e08b0fe0-7bfa-4932-94bf-c3f95412692b">

I'll try to force-cast the subsequent type

Reply:
Execute program's TS course has this issue considered.

They would recommend something like:
```js
} catch ( e: unknown ) {
    if ( e instance of DSPError ) {
        return await ...
    } else {
        throw new Error( ... ):
    }
```